tests/generators/test_azure.py

```diff
@@ -0,0 +1,47 @@
+import os
+import pytest
+
+import garak.cli
+from garak.generators.azure import AzureOpenAIChat
+
+DEFAULT_GENERATIONS_QTY = 10
+
+@pytest.mark.skipif(
+    os.getenv("AZURE_DEPLOYMENT_NAME", None) is None,
+    reason=f"AZURE_DEPLOYMENT_NAME env variable is not set",
+)
+
+@pytest.mark.skipif(
+    os.getenv("AZURE_ENDPOINT", None) is None,
+    reason=f"AZURE_ENDPOINT env variable is not set",
+)
+
+@pytest.mark.skipif(
+    os.getenv(AzureOpenAIChat.ENV_VAR, None) is None,
+    reason=f"Azure OpenAI API key is not set in {AzureOpenAIChat.ENV_VAR}",
+)
```

Comment:
Instead of skipping all tests add a fixture that will provide the required configuration values.

A separate test can call `generate` or consider using the mock response fixtures as done here:
https://github.com/leondz/garak/blob/c6abfabe05a809617c5f726a8c6ab2b9436a27ac/tests/generators/test_openai.py#L34-L36

Reply:
I was not able to completely improve that in the last commit, I added a test case to test invalid model names like for `test_openai.py` but mocking a request (and so fake environment variables) doesn't seem to be required, there is no request involved? 

The other test case, about `generate `, is still skipped if the environment variables are not set, it seems consistent with what is done in other tests. It's not easy to mock LLM generations.